### ISSUE TITLE
When renaming a non-ignored file to a name that matches the "Ignore Regex Rule", the title still gets synchronized

### ISSUE THREAD

**Issue by fkuersch** [dvcrn/obsidian-filename-heading-sync#64]
### Steps to Reproduce:
- Set the "Ignore Regex Rule" to "^\+.*" to match files starting with +
- Create a new file
- Set a h1, ie. # example title
  - The plugin renames the file to "example title"
- Change the file name to "+shouldbeignored"

### Expected behavior:
- The h1 should stay the same ("example title")

### Actual behavior:
- The h1 gets renamed to "+shouldbeignored"

### Possible solution:
The current [handleSyncFilenameToHeading](https://github.com/dvcrn/obsidian-filename-heading-sync/blob/0c0a4391acd8901e97cbf577cd70a0290464f846/main.ts#L182) checks [for the old path only](https://github.com/dvcrn/obsidian-filename-heading-sync/blob/0c0a4391acd8901e97cbf577cd70a0290464f846/main.ts#L197). It should also check whether the new file name is ignored.

**Comment by dvcrn** [comment]
Yeah currently it only looks at the current file name. If added, this should probably behind an option because I can imagine some people don't want this

**Comment by fkuersch** [comment]
Since I've only been using this plugin for a few days by now, I find it hard to think of a situation where you would want an option for that. There might be use cases for that option, but I just cannot imagine a specific situation.

After a little thought, i want to put it a little more broadly:

**Current behavior:**
Change from ignored to non-ignored -> don't sync
Change from non-ignored to ignored -> sync

**Expected behavior:**
Change from ignored to non-ignored -> sync
Change from non-ignored to ignored -> don't sync

That is in each case in both directions (h1 -> filename and filename -> h1)
